### ISSUE TITLE
Installing Prophet R-Package in Databricks - Is there a faster way?

### ISSUE THREAD

**Issue by nknauer** [facebook/prophet#1857]
Hi I am trying to install prophet r-package in a Databricks notebook. 

Is there a faster way to do this than below? It takes 25min to install the prophet package this way.

```
Sys.setenv(DOWNLOAD_STATIC_LIBV8 = 1)
remotes::install_github("jeroen/V8")
devtools::install_version("rstantools", version = "2.0.0")
install.packages('prophet')
```

**Comment by bletham** [comment]
What is taking most of the time? My experience is that installing rstan and installing dplyr both take a very long time because there is a lot of C++ code to compile. The actual installation of prophet should take about 1-2 minutes.

**Comment by nknauer** [comment]
The dependency packages like V8 and rstantools are not the bottleneck unfortunately. Instead it is the prophet package itself.
However within the prophet package there are dependencies as is to make it run such as: 
`checkmate, matrixStats, zoo, inline, loo, dygraphs, extraDistr, RcppParallel, rstan, StanHeaders, xts, RcppEigen`

Is there a better way to install prophet within Databricks? Seems unavoidable at this point it will take 20-25min each time I run my notebook

Here is a screenshot of all the load times by line item in Databricks:
![image](https://user-images.githubusercontent.com/23511145/116823709-e5256300-ab53-11eb-8bc4-826fb6b4fd9c.png)
![image](https://user-images.githubusercontent.com/23511145/116823714-f2dae880-ab53-11eb-9ac7-cf15534e668d.png)
![image](https://user-images.githubusercontent.com/23511145/116825979-bb723900-ab5f-11eb-86e9-6c534e324635.png)

**Comment by bletham** [comment]
Yeah the issue is in all of those dependencies, which unfortunately are actually needed and so there isn't a workaround to skip some of them. I think this issue goes a bit beyond my understanding of both R packaging and databricks, so hopefully someone else will be able to chime in, but ultimately I think you'd need to somehow have the dependencies included in the image so that they don't have to be installed from scratch.

**Comment by nknauer** [comment]
Very interesting! Never tried something like this before. I will attempt this and get back to you with results. Could this method also work with installing it directly to a cluster?

**Comment by dwh1142** [comment]
I don't think it would work for a cluster scoped package (since that functionality just installs using `install.package` from CRAN).  Also, the idea of renv, is to separate environments based on the project you are working on, so it could be argued you don't want one environment installed on the entire cluster.  But if you set your global package cache environment variable in an init script, and install `renv` as a cluster scoped package, it will be very fast to restore environments once you cluster is spun up

**Comment by nknauer** [comment]
Thanks, will try out the init script. I also asked on stackoverflow too https://stackoverflow.com/questions/69884057/installing-remote-r-package-to-databricks-cluster-rather-than-notebook
